lesson17/main.js

```diff
@@ -0,0 +1,123 @@
+  const createdElement = document.createElement(element);
+  createdElement.classList.add(name);
+  return createdElement;
+};
+
+const createLoader = () => {
+  const loader = document.createElement("div");
+  const loaderImage = document.createElement("img");
+  loader.id = "js-loader";
+  loaderImage.src = "img/loading-circle.gif";
+  slideshowWrap.appendChild(loader).appendChild(loaderImage);
+};
+
+const loading = () => {
+  const loader = document.getElementById("js-loader");
+  loader.classList.add("loading");
+};
+
+const loaded = () => {
+  const loader = document.getElementById("js-loader");
+  loader.classList.add("loaded");
+};
+
+const numOfSecond = 3000;
+const fetcheDataInSecond = (sec, jsonUrl) => {
+  return new Promise((resolve, reject) => {
+    setTimeout(() => {
+      resolve(fetch(jsonUrl));
+    }, sec);
+  });
+};
+
+const fetcheImgData = async () => {
+  try {
+    const response = await fetcheDataInSecond(numOfSecond, jsonUrl);
+    const json = await response.json();
+    return json.images;
+  } catch (error) {
+    slideshowWrap.textContent = "データの取得ができませんでした。";
+    console.error(error);
+  } finally {
+    loaded();
+  }
+};
+
+const createListsOfImg = (imgData) => {
+  const fragment = document.createDocumentFragment();
+  imgData.forEach((imgVal, index) => {
+    const li = createElementWithClassName("li", "slideshow__img");
+    const img = document.createElement("img");
+    img.src = imgVal.src;
+
+    //とりあえず最初のインデックスを表示
+    index === 0 && li.classList.add("is-show");
+
+    fragment.appendChild(li).appendChild(img);
+  });
+  ul.appendChild(fragment);
+};
+
+const createArrowBtnForSlideshow = () => {
+  createNextBtn();
+  createBackBtn();
+};
+
+const createNextBtn = () => {
+  const nextBtn = createElementWithClassName("button", "btn");
+  nextBtn.id = "js-next-btn";
+  nextBtn.textContent = ">";
+  ul.insertAdjacentElement("afterend", nextBtn);
+  nextBtn.addEventListener(
+    "click",
+    () => {
+      goNext();
+    },
+    false
+  );
+};
+
+const createBackBtn = () => {
+  const backBtn = createElementWithClassName("button", "btn");
+  backBtn.id = "js-back-btn";
+  backBtn.textContent = "<";
+  ul.insertAdjacentElement("beforebegin", backBtn);
+  backBtn.addEventListener(
+    "click",
+    () => {
+      goBack();
+    },
+    false
+  );
+};
+
+const goNext = () => {
+  const currentImg = document.querySelector(".is-show");
+  if (currentImg.nextElementSibling) {
+    currentImg.classList.remove("is-show");
+    currentImg.nextElementSibling.classList.add("is-show");
+  }
+};
+
+const goBack = () => {
+  const currentImg = document.querySelector(".is-show");
+  if (currentImg.previousElementSibling) {
+    currentImg.classList.remove("is-show");
+    currentImg.previousElementSibling.classList.add("is-show");
+  }
+};
+
+const init = async () => {
+  createLoader();
+  loading();
+  const imgData = await fetcheImgData();
+  createListsOfImg(imgData);
+  createArrowBtnForSlideshow();
+};
```

Comment:
IMO
About `await fetcheImgData();`.
I think I can handle it with try-catch here too.

The following is what I learned from Morita-san's feedback when I took on this challenge.

If `imgData.length ! == 0`.

```js
createListsOfImg(imgData);
createArrowBtnForSlideshow();
```

It was reviewed that it would be better to display ``data is empty`` otherwise.

Reply:
Thank you  for teaching!

At first I thought it might be doing the same thing, since try-catch is also used in the function `fetcheImgData` (if the function `fetcheImgData` fails in any way, try-catch in this function will tell me the error in the catch part).
However, I think this try-catch corresponds to the case when `await fetcheImgData();` is undefined :) 💡 

Is this idea correct? 
[f13e153](https://github.com/yuka830/js-lessons/pull/21/commits/f13e15374ff7b122a9b1155b38ad8caab62e3ccd)

Reply:
This part is written in case something unexpected happens. I think yuka-san's interpretation is fine.

I thought it would be safer to write in constants here.
```js
  const imgData = await fetcheImgData()
    .catch((e) => console.error(e))
    .finally(() => console.log("処理が完了しました。"));
```

Reply:
Thanks for sharing. I didn't know that too!
I had to rewrite it as let because const would give an error because it was not initialized, but this is safer!
I'll keep this in mind for future reference :) Thank you!